massa-network-worker/src/node_worker.rs

```diff
@@ -207,74 +185,19 @@ impl NodeWorker {
-                                massa_trace!(
-                                    "node_worker.run_loop. receive Message::AskForOperations: ",
-                                    {"node": self.node_id, "operation_ids": operation_prefix_ids}
-                                );
-                                //massa_trace!("node_worker.run_loop. receive Message::AskForOperations", {"node": self.node_id, "operations": operation_ids});
-                                self.send_node_event(NodeEvent(self.node_id, NodeEventType::ReceivedAskForOperations(operation_prefix_ids))).await;
-                            }
-                            Message::OperationsAnnouncement(operation_prefix_ids) => {
-                                massa_trace!("node_worker.run_loop. receive Message::OperationsBatch", {"node": self.node_id, "operation_prefix_ids": operation_prefix_ids});
-                                self.send_node_event(NodeEvent(self.node_id, NodeEventType::ReceivedOperationAnnouncements(operation_prefix_ids))).await;
-                            }
-                            Message::Endorsements(endorsements) => {
-                                massa_trace!("node_worker.run_loop. receive Message::Endorsement", {"node": self.node_id, "endorsements": endorsements});
-                                self.send_node_event(NodeEvent(self.node_id, NodeEventType::ReceivedEndorsements(endorsements))).await;
-                            }
-                            _ => {
-                                // TODO: Write a more user-friendly warning/logout after several consecutive fails? see #1082
-                                massa_trace!("node_worker.run_loop.self.socket_reader.next(). Unexpected message Warning", {});
-                            },
+                res = &mut node_reader_handle => {
+                    reader_joined = true;
+                    _exit_reason_reader = match res {
+                        Ok(r) => {
+                            r
+                        },
+                        Err(e) => {
+                            debug!("node_worker.run_loop.node_reader.error: {}", e);
+                            ConnectionClosureReason::Failed
                         }
-                    },
-                    Ok(None)=> {
-                        massa_trace!("node_worker.run_loop.self.socket_reader.next(). Ok(None) Error", {});
-                        break
-                    }, // peer closed cleanly
-                    Err(err) => {  // stream error
-                        massa_trace!("node_worker.run_loop.self.socket_reader.next(). receive error", {"error": format!("{}", err)});
-                        exit_reason = ConnectionClosureReason::Failed;
-                        break;
-                    },
-                },
-
+                    };
+                    break;
+                }
                 // node command
                 cmd = self.node_command_rx.recv() => {
```

Comment:
To reduce contention, I recommend not managing this here but directly providing writer_command_tx upstream so that the network main loop communicates directly with the node's writer task

Reply:
Done